packages/text-field/src/TextField.js

```diff
@@ -90,123 +87,23 @@ export default class TextField extends Component {
-      this.state.value.length > 0
-    );
-  }
-
-  shouldShowInstructions() {
-    if (this.props.instructions) {
-      if (this.props.errors) {
-        return !this.props.hideInstructionsOnErrors;
-      }
-      return true;
-    }
-
-    return false;
-  }
-
   render() {
-    const hasClearableInput = this.hasClearableInput();
-
     return (
-      <div
-        className={cx("hig__text-field", {
-          "hig__text-field--required": this.props.required,
-          "hig__text-field--clear-button-visible": hasClearableInput
-        })}
+      <TextFieldController
+        defaultValue={this.props.defaultValue}
+        onChange={this.props.onChange}
+        onClearButtonClick={this.props.onClearButtonClick}
+        value={this.props.value}
       >
-        <div
-          className={cx("hig__text-field__content", {
-            "hig__text-field__content--with-icon": this.props.icon
-          })}
-        >
-          <div className="hig__text-field__input-wrapper">
-            {this.props.icon && (
-              <label
-                className={cx("hig__text-field__icon-v1", {
-                  "hig__text-field__icon-v1--disabled": this.props.disabled
-                })}
-                htmlFor={this.props.id}
-              >
-                {this.props.icon}
-              </label>
-            )}
-
-            <Input
-              id={this.props.id}
-              value={this.state.value}
-              onChange={this.handleChange}
-              name={this.props.name}
-              type={this.props.type}
-              disabled={this.props.disabled}
-              placeholder={this.props.placeholder}
-              onBlur={this.props.onBlur}
-              onFocus={this.props.onFocus}
-              onInput={this.props.onInput}
-            />
-
-            {this.props.label && (
-              <label
-                htmlFor={this.props.id}
-                className="hig__text-field__label-v1"
-              >
-                {this.props.label}
-              </label>
-            )}
-
-            {hasClearableInput && (
-              <span className="hig__text-field__clear">
-                <IconButton
-                  type="transparent"
-                  icon="clear-small"
-                  title="Clear field"
-                  onClick={this.handleInputClear}
-                />
-              </span>
-            )}
-          </div>
-
-          {this.shouldShowInstructions() && (
-            <p className="hig__text-field__instructions">
-              {this.props.instructions}
-            </p>
-          )}
-
-          {this.props.errors && (
-            <p className="hig__text-field__errors">{this.props.errors}</p>
-          )}
-
-          {this.props.required && (
-            <p className="hig__text-field__required-notice">
-              {this.props.required}
-            </p>
-          )}
-        </div>
-      </div>
+        {({ value, onChange, onClearButtonClick }) => (
```

Comment:
Lets keep the handler names starting with `handle`. Typically, only the prop names on the components themselves start with `on`. 

```js
<Controller>
  {({ handleClick }) => (
    <div onClick={handleClick} />
  )} 
</Controller>
```